src/main/java/com/noxag/newnox/textanalyzer/algorithms/BibliographyAnalyzer.java

```diff
@@ -1,18 +1,111 @@
 package com.noxag.newnox.textanalyzer.algorithms;
 
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
 
 import org.apache.pdfbox.pdmodel.PDDocument;
 
 import com.noxag.newnox.textanalyzer.TextanalyzerAlgorithm;
 import com.noxag.newnox.textanalyzer.data.Finding;
+import com.noxag.newnox.textanalyzer.data.TextFinding;
+import com.noxag.newnox.textanalyzer.data.TextFinding.TextFindingType;
+import com.noxag.newnox.textanalyzer.data.pdf.PDFPage;
+import com.noxag.newnox.textanalyzer.data.pdf.TextPositionSequence;
+import com.noxag.newnox.textanalyzer.util.PDFTextExtractionUtil;
+import com.opencsv.CSVReader;
 
 public class BibliographyAnalyzer implements TextanalyzerAlgorithm {
+    private static final Logger LOGGER = Logger.getLogger(CommonAbbreviationAnalyzer.class.getName());
+    private static final String BIBLIOGRAPHY_IDENTIFICATION_LIST_PATH = "src/main/resources/analyzer-conf/bibliography-identifications.csv";
 
     @Override
     public List<Finding> run(PDDocument doc) {
-        // TODO Auto-generated method stub
-        return null;
+        List<Finding> findings = new ArrayList<>();
+        try {
+            findings.addAll(getReferencesWithoutSource(doc));
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Could not strip text from document", e);
+        }
+        return findings;
+    }
+
+    private List<Finding> getReferencesWithoutSource(PDDocument doc) throws IOException {
+        List<Finding> findings = new ArrayList<>();
+        List<PDFPage> pages = PDFTextExtractionUtil.extractText(doc);

```

Comment:
Könntest du bitte das auslesen des PDDocs auslagern in Zeile 31 und die getReferencesWithoutSource methode dann mit einem List<PDFPage> parameter erst in Zeile 34/35 aufrufen

Reply:
Done